cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -184,23 +185,28 @@ func runControlPlanePrepareControlPlaneSubphase(c workflow.RunData) error {
 		return err
 	}
 
-	fmt.Printf("[control-plane] Using manifest folder %q\n", kubeadmconstants.GetStaticPodDirectory())
+	fmt.Printf("[control-plane] Using manifest folder %q\n", data.ManifestDir())
+
+	// If we're dry-running, set CertificatesDir to default value to get the right cert path in static pod yaml
+	if data.DryRun() {
+		cfg.CertificatesDir = filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.DefaultCertificateDir)
+	}
```

Comment:
but during dryrun the directory where the certs were written is now data.CertificateWriteDir().
are you getting errors if this is not added here?

Reply:
It won't get errors here, but if I don't change it, it will write tmp cert dir path to static pod yaml instead of "/etc/kubernetes/pki/". 
And If I remake [CreateStaticPodFiles()](https://github.com/kubernetes/kubernetes/blob/f4e7828674d1526f2b04def99c20242f1548c2d8/cmd/kubeadm/app/phases/controlplane/manifests.go#L95) to accept path directly, [GetStaticPodSpecs](https://github.com/kubernetes/kubernetes/blob/f4e7828674d1526f2b04def99c20242f1548c2d8/cmd/kubeadm/app/phases/controlplane/manifests.go#L49) need to do the same and other places that use them. 
any suggestion? @neolit123

Reply:
making these functions accept a directory seems like a good idea for dry-run / testing?